File: leet-code/215. Kth Largest Element in an Array/KthLargestElementInAnArray215.java
package com.example;

import java.util.Collections;
import java.util.PriorityQueue;
import java.util.Queue;
import java.util.stream.IntStream;

public class KthLargestElementInAnArray215 {
    public int findKthLargest(int[] nums, int k) {
        Queue<Integer> queue = new PriorityQueue<>(Collections.reverseOrder());
        for (int num : nums) queue.add(num);
        IntStream.range(0, k).forEach(i -> queue.poll());
        return queue.poll();
    }
}

File: leet-code/215. Kth Largest Element in an Array/KthLargestElementInAnArray215.kt
package com.example

class Solution {
    fun findKthLargest(nums: IntArray, k: Int): Int {
        val pq = java.util.PriorityQueue<Int>(java.util.Collections.reverseOrder())
        for (num in nums) {
            pq.add(num)
        }
        repeat(k - 1) { pq.poll() }
        return pq.poll()
    }
}
